usb/core.py

```diff
@@ -756,6 +758,22 @@ def __init__(self, dev, backend):
         else:
             self.port_number = None
 
+        if getattr(desc, 'speed', None):
+            self.iSpeed = int(desc.speed)
+            if self.iSpeed == 1:
+                self.speed = 'low'
+            elif self.iSpeed == 2:
+                self.speed = 'full'
+            elif self.iSpeed == 3:
+                self.speed = 'high'
+            elif self.iSpeed == 4:
+                self.speed = 'super'
+            else:
+                self.speed = 'unknown'
```

Comment:
You can define the constants:
```python
SPEED_LOW = 1
SPEED_FULL = 2
SPEED_HIGH = 3
SPEED_SUPER = 4
SPEED_UNKNOWN = -1
```

And then just assign the value:

```python
self.speed = desc.speed
```

A string representation of the speed can be achieved through the implementation of a translation function in the [`util`](https://github.com/walac/pyusb/blob/master/usb/util.py) module.